test/live_queue_test.exs

```diff
@@ -0,0 +1,185 @@
+defmodule Membrane.VideoCompositor.LiveQueueTest do
+  @moduledoc false
+  use ExUnit.Case
+
+  alias Membrane.{Buffer, RawVideo, Time, VideoCompositor}
+
+  alias Membrane.VideoCompositor.{
+    BaseVideoPlacement,
+    CompositorCoreFormat,
+    Scene,
+    SceneChangeEvent,
+    VideoConfig
+  }
+
+  alias Membrane.VideoCompositor.Queue.Live, as: LiveQueue
+
+  alias Membrane.VideoCompositor.Support.Handler
+
+  @video_config %VideoConfig{
+    placement: %BaseVideoPlacement{
+      position: {0, 0},
+      size: {1.0, 1.0}
+    }
+  }
+
+  @video_stream_format %RawVideo{
+    width: 1920,
+    height: 1080,
+    framerate: {1, 1},
+    pixel_format: :I420,
+    aligned: true
+  }
+
+  @pad1 {Pad, :input, 1}
+  @pad2 {Pad, :input, 2}
+
+  @pad1_frame <<1>>
+  @pad2_frame <<2>>
+
+  @pad2_pts_offset Time.seconds(1) + 1
+
+  test "if sets up correctly" do
+    setup_videos()
+  end
+
+  test "if sends correct scene and stream format actions before buffers" do
+    state = setup_videos() |> send_both_pads_frames()
+
+    stream_format = %CompositorCoreFormat{
+      pad_formats: %{@pad1 => @video_stream_format, @pad2 => @video_stream_format}
+    }
+
+    stream_format_action = {:stream_format, {:output, stream_format}}
+
+    scene = %Scene{video_configs: %{@pad1 => @video_config, @pad2 => @video_config}}
+    scene_action = {:event, {:output, %SceneChangeEvent{new_scene: scene}}}
+
+    buffer = %Buffer{payload: %{@pad1 => @pad1_frame, @pad2 => @pad2_frame}, pts: 0, dts: 0}
+    buffer_action = {:buffer, {:output, buffer}}
+
+    {actions, _state} = LiveQueue.handle_tick(:buffer_scheduler, %{}, state)
+    assert ^stream_format_action = Enum.at(actions, 0)
+    assert ^scene_action = Enum.at(actions, 1)
+    assert ^buffer_action = Enum.at(actions, 2)
+  end
+
+  test "if sends stream format and scene only once" do
```

Comment:
I don't think it anyhow checks if the stream format is sent only once

Reply:
Fixed